<a href="https://colab.research.google.com/github/blacurly/sharing-github-oliv/blob/main/EDA_Test_Wine.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Memahami Exploratory Data Analysis menggunakan Wine_datasets bersama Dphi Tech**

Kali ini, kita akan coba melakukan EDA menggunakan Wine_datasets, sebelum melanjutkan ke analisa lanjutan perihal memprediksi kualitas Wine yang terbaik.

In [3]:
import numpy as np #Package dasar untuk aljabar linear dan multidomensional arrays, dapat juga menghitung mean, median
import pandas as pd #digunakan untuk data processing dan kerja dengan CSV files dengan DataFrames, bisa menjadi tool data analisis dan manipulasi


In [4]:
#Loading dataset
data_eda = pd.read_csv("https://raw.githubusercontent.com/dphi-official/Datasets/master/Wine_Dataset/winequality-red.csv", sep=";")

In [5]:
#Melakukan review awal, untuk memahami feeling inisial tentang data yang akan iolah
#head() fungsi panda untuk memanggil 5 data pertama di datasets
#tails() fungsi panda untuk memanggil 5 data terakhir di datasets
data_eda.head()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
0            7.4              0.70         0.00  ...       0.56      9.4        5
1            7.8              0.88         0.00  ...       0.68      9.8        5
2            7.8              0.76         0.04  ...       0.65      9.8        5
3           11.2              0.28         0.56  ...       0.58      9.8        6
4            7.4              0.70         0.00  ...       0.56      9.4        5

[5 rows x 12 columns]

In [6]:
data_eda.tail()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
1594            6.2             0.600         0.08  ...       0.58     10.5        5
1595            5.9             0.550         0.10  ...       0.76     11.2        6
1596            6.3             0.510         0.13  ...       0.75     11.0        6
1597            5.9             0.645         0.12  ...       0.71     10.2        5
1598            6.0             0.310         0.47  ...       0.66     11.0        6

[5 rows x 12 columns]

In [7]:
#Shape: Mengecek total baris dan kolom dari datasets
data_eda.shape

(1599, 12)

Observasi:
terdapat 1599 baris dengan 12 kolom yang berperan sebagai variabel/features


**Get A statistical Overwiew using Describe()**


In [8]:
#Describe() merupakan fungsi di pandas untuk mendapatkan ringkasan statistik yang mengandung Mean, sd, min, max serta data quantile
data_eda.describe()

fixed acidity  volatile acidity  ...      alcohol      quality
count    1599.000000       1599.000000  ...  1599.000000  1599.000000
mean        8.319637          0.527821  ...    10.422983     5.636023
std         1.741096          0.179060  ...     1.065668     0.807569
min         4.600000          0.120000  ...     8.400000     3.000000
25%         7.100000          0.390000  ...     9.500000     5.000000
50%         7.900000          0.520000  ...    10.200000     6.000000
75%         9.200000          0.640000  ...    11.100000     6.000000
max        15.900000          1.580000  ...    14.900000     8.000000

[8 rows x 12 columns]

**Observation**


*   Dapat dilihat bahwa mean < median pada masing-masing kolom. Median dapa direpresentasikan dengan 50% atau percentile ke-50. Ini mengindikasikan adanya outlier. contoh: kita punya dataset yang berisi nilai 30,31,32, dan 2. Maka mean = 23.75 < median (30.5), hal tersebut terdampak dari data ekstrem yang bernilai 2
* Ada perbedaan besar antara percentile ke-75 dan nilai maks dari prediktor "Residual Sugar", "free sulfur dioxide", "total sulfur dioxide". Ini menunjukkan bahwa beberapa nilai dari 3 variabel ini terletak lebih jauh dari range nilai umum (hingga persentil ke-75)
* Dengan demikian, observasi 1 dan 2 menunjukkan bahwa ada nilai ekstrim berupa outlier dalam datasets.  



**Exploring the Features/Variable**


In [9]:
#Mengenal atak mengeksplor features/kolom dari datasets
data_eda.columns

Index(['fixed acidity', 'volatile acidity', 'citric acid', 'residual sugar',
       'chlorides', 'free sulfur dioxide', 'total sulfur dioxide', 'density',
       'pH', 'sulphates', 'alcohol', 'quality'],
      dtype='object')

Setelah itu, mari kita lihat feature mana yang akan berperan sebagai **target variable** dan mana **input variabel**

Di sini *quality* adalah variabel target karena kami mencoba mengetahui antara dua jenis anggur, jenis yang manakah memiliki kualitas yang lebih baik. Sedangkan, variabel input kita adalah: *fixed acidity*, *volatile acidity*, *citric acid*, *residual sugar*, *chlorides*, *free sulfur dioxide*, *total sulfur dioxide*, *density*,*pH*, *sulphates*, and *alcohol*. Semua variabel ini dapat membantu kita untuk memprediksi kualitas wine


Setelah loading data, perlu untuk memahami apakah semua data cocok dilibatkan dalam model.

**Melihat Unique Value dari Variabel Target (Quality)**


In [10]:
data_eda['quality'].unique()

array([5, 6, 7, 4, 8, 3])

dari hasil diatas, dapat disimpulkan beberapa hal yaitu:

*   Variabel target sifatnya discrete dan categorical
*   Skala scor *quality* berada pada kisaran 1-10 (1 jelas  -> 10 Terbaik)
*   Hanya ada rating/skor 3-8 sedangkan 1,2,9,10 tidak ada


In [11]:
data_eda['quality'].value_counts()

5    681
6    638
7    199
4     53
8     18
3     10
Name: quality, dtype: int64

*   Hasilnya menunjukan jumlah frekuensi dari skor *quality* ada pada descending order atau urutannmenurun
*   nilai *quality* sebagian besar skorratingnya pada kategori 5,6,7
*   Hanya beberapa data/observasi pada kategori skor 3 dan 8





**Mengecek dan Mengatasi Missiong Value**

Menangani missing value adalah bagian yang penting pada proses preparation dan cleaning data, karena hampir semua data di dunia nyata mengandung missing value.Di *pandas* terdapat fungsi *isnull(), isna()* untuk mendeteksi missing value.

In [12]:
data_eda.isna().sum() #sum merujuk pada total missing value di masing2 features

fixed acidity           0
volatile acidity        0
citric acid             0
residual sugar          0
chlorides               0
free sulfur dioxide     0
total sulfur dioxide    0
density                 0
pH                      0
sulphates               0
alcohol                 0
quality                 0
dtype: int64

In [13]:
data_eda.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1599 non-null   float64
 1   volatile acidity      1599 non-null   float64
 2   citric acid           1599 non-null   float64
 3   residual sugar        1599 non-null   float64
 4   chlorides             1599 non-null   float64
 5   free sulfur dioxide   1599 non-null   float64
 6   total sulfur dioxide  1599 non-null   float64
 7   density               1599 non-null   float64
 8   pH                    1599 non-null   float64
 9   sulphates             1599 non-null   float64
 10  alcohol               1599 non-null   float64
 11  quality               1599 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 150.0 KB




*   data hanya mengandung nilai float dan integer
*   Tidak ada missing value

